lib/tasks/load_legacy_mappings.rake

```diff
@@ -0,0 +1,29 @@
+desc "Populate legacy_taxons links for each topic taxon according to a CSV"
+task :load_legacy_mappings, [:csv_path] => :environment do |_t, args|
+  mapping = Hash.new { |h, k| h[k] = [] }
+
+  CSV.foreach(args[:csv_path]) do |row|
+    mapping[row[1]] << row[0].match(/.+\[(.+)\]/)[1]
+  end
+
+  api = Services.publishing_api
+  api.client.options[:timeout] = 30
+
+  mapping.each do |topic_taxon, legacy_taxons|
+    begin
+      topic_taxon_id = api.lookup_content_id(base_path: topic_taxon)
+      raise "Lookup failed for #{topic_taxon}" unless topic_taxon_id
+      topic_taxon_links = api.get_links(topic_taxon_id)["links"]
+
+      legacy_taxon_ids = api.lookup_content_ids(base_paths: legacy_taxons)
+      missing_legacy_taxons = legacy_taxons - legacy_taxon_ids.keys
+      raise "Lookup failed for #{missing_legacy_taxons}" if missing_legacy_taxons.any?
+      topic_taxon_links["legacy_taxons"] = legacy_taxon_ids.values
+
+      puts "Adding #{legacy_taxons.count} legacy taxons for #{topic_taxon}"
+      api.patch_links(topic_taxon_id, links: topic_taxon_links, bulk_publishing: true)
```

Comment:
The Publishing API supports "optimistic locking" for these kind of requests. I don't think this is something we'd want to handle explicitly here, but I was wondering if we could add some block based thing that would handle fetching the links and changing them, while handling the `previous_version` thing to ensure that there are not any conflicts. Something like...
```
api.update_links(topic_taxon_id, bulk_publishing: true) do |links|
  links["legacy_taxons"] = legacy_taxon_ids.values
end
```
What do you think? I don't see this as something that is necessary to solve now, but it would be good to completely avoid the possibility of overriding any changes made concurrently in Content Tagger when running this script.

Reply:
I forgot the link to the docs https://docs.publishing.service.gov.uk/apis/publishing-api/api.html#optimistic-locking-previous_version

Reply:
I think supporting concurrent transactions is a good habit, so why not...done.